cpp/src/arrow/io/memory.cc

```
@@ -299,14 +301,16 @@ Status BufferReader::Tell(int64_t* position) const {
   return Status::OK();
 }
 
-util::string_view BufferReader::Peek(int64_t nbytes) const {
+Status BufferReader::Peek(int64_t nbytes, util::string_view* out) {
   if (!is_open_) {
-    return {};
+    *out = {};
+    return Status::OK();
```

Should return `Status::Invalid` like other methods.

---

Same question as above

---

Same answer: not returning a Status meant we couldn't fail explicitly :-)

---

done